src/JerryDebuggerSerialClient.ts

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2019-present Samsung Electronics Co., Ltd. and other contributors
+ * Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as SerialPort from 'serialport';
+
+export interface JerryDebuggerSerialOptions {
+  delegate: JerryDebuggerDelegate;
+  serialConfig?: string;
+}
+
+export interface JerryDebuggerDelegate {
+  onMessage: (message: Uint8Array) => void;
+  onClose?: () => void;
+}
+
+export class JerryDebuggerSerialClient {
+  readonly serialConfig: string;
+  readonly channel: string;
+  private serialport?: SerialPort;
+  private connectPromise?: Promise<void>;
+  private delegate: JerryDebuggerDelegate;
+  private buffer: ArrayBuffer;
+
+  constructor(options: JerryDebuggerSerialOptions) {
+    this.delegate = options.delegate;
+    this.serialConfig = options.serialConfig;
+    this.buffer = new Uint8Array();
+    this.channel = 'serial';
+  }
+
+  public async connect(): Promise<void> {
+    if (this.connectPromise) {
+      return this.connectPromise;
+    }
+
+    let portsList = [];
+
+    await SerialPort.list().then(
+      ports => ports.forEach((port) => {
+        portsList.push(port.comName);
+      })
+    );
+
+    const config = this.serialConfig.split(',', 5);
+
+    this.connectPromise = new Promise((resolve, reject) => {
+      if (config.length !== 5) {
+        reject(new Error('Invalid serial configuration!'));
+        return;
+      }
+
+      let openOptions: SerialPort.OpenOptions = {};
+      const portID = config[0];
+      if (!portsList.includes(portID)) {
+        reject(new Error(`Invalid portID (${portID}) for serial configuration!`));
+      }
+
+      openOptions.baudRate = parseInt(config[1], 10);
+
+      switch (config[2]) {
+        case '5':
+          openOptions.dataBits = 5;
+          break;
+        case '6':
+          openOptions.dataBits = 6;
+          break;
+        case '7':
+          openOptions.dataBits = 7;
+          break;
+        case '8':
+          openOptions.dataBits = 8;
+          break;
+        default:
+          reject(new Error(`Invalid data bit (${config[2]}) for serial configuration`));
```

Comment:
Is this a sentence? (There is no dot ot exclamation mark :)) Please fix other sentences in this patch.